frameworks/sveltekit.md

```diff
@@ -13,7 +19,55 @@ For `Type declarations`, `Prompt for update` and `Periodic SW Updates` go to [Sv
 :::
 
 ::: tip
-You should remove all references to [SvelteKit service worker module](https://kit.svelte.dev/docs/modules#$service-worker) to disable it on your application.
+If you're using `0.1.*` version of `SvelteKitPWA`, you should remove all references to [SvelteKit service worker module](https://kit.svelte.dev/docs/service-workers) to disable it on your application.
+:::
+
+## Generate Custom Service Worker
+
+From version `0.2.0`, `SvelteKitPWA` plugin will delegate your custom service worker build to `SvelteKit`, and so, you can use `service-worker` as the name of your service worker file (VanillaJS or TypeScript).
+You will need to exclude the service worker registration from the `SvelteKit` configuration if you're using any virtual module:
+```js
+// svelte.config.js
+/** @type {import('@sveltejs/kit').Config} */
+const config = {
+  kit: {
+    serviceWorker: {
+      register: false
+    }
+  }
+};
+```
+
+You can also configure a custom service worker name, `SvelteKitPWA` plugin will rename your service worker (remember to also update the service worker name in `SvelteKitPWA` plugin configuration):
+```js
+// svelte.config.js
+/** @type {import('@sveltejs/kit').Config} */
+const config = {
+  kit: {
+    files: {
+      serviceWorker: 'src/prompt-sw.ts',
+    }
+  }
+};
+```
+
+You can check SvelteKit docs for more information about [Service Worker](https://kit.svelte.dev/docs/service-workers).
+
+::: warning
+If your custom service working is importing any `workbox-*` module (`workbox-routing`, `workbox-strategies`, etc), you will need to hack Vite build process in order to remove non `ESM` special replacements from the build process (if you don't include `process.env.NODE_ENV`, the service worker will not be registered). You only need to add this entry in your Vite config file:
```

Comment:
this is perhaps not required with Vite 4.2? https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md#420-beta0-2023-02-27

Reply:
If using Vite 4.1 we need to "hack" Vite build. 

Maybe we can keep it, and add a suggestion when Vite 4.2 released. 

I will check it using the latest beta once new minor version (0.2.0) released and published.

Reply:
Testing with Vite 4.2.0-beta.1 not working using injectManifest strategy, `prompt-sw.js` containing `process.env.NODE_ENV` entries.